packages/react-native-sdk/src/storage/asyncStorage.ts

```diff
@@ -1,58 +1,55 @@
 import AsyncStorage from '@react-native-async-storage/async-storage';
 
-async function getTable(dbName: string, tableName: string): Promise<Map<string, any>> {
-  const data = await AsyncStorage.getItem(`${dbName}.${tableName}`);
-  if (data) {
-    return new Map(JSON.parse(data));
-  }
-  return new Map<string, any>();
+const SEPARATOR = ':';
+
+function getKey(dbName: string, tableName: string, id: string) {
+  return `${dbName}${SEPARATOR}${tableName}${SEPARATOR}${id}`;
 }
 
-function setTable(dbName: string, tableName: string, table: Map<string, any>): Promise<void> {
-  return AsyncStorage.setItem(`${dbName}.${tableName}`, JSON.stringify([...table]));
+async function getTableKeys(dbName: string, tableName: string) {
+  const allKeys = await AsyncStorage.getAllKeys();
+  return allKeys.filter(key => key.indexOf(`${dbName}${SEPARATOR}${tableName}${SEPARATOR}`) === 0);
 }
 
 export async function find(dbName: string, tableName: string) {
-  const table = await getTable(dbName, tableName);
-  return Array.from(table.values());
+  const tableRecords = await AsyncStorage.multiGet(await getTableKeys(dbName, tableName));
+  return tableRecords.map(([key, value]) => JSON.parse(value));
 }
 
 export async function count(dbName: string, tableName: string) {
-  const docs = await find(dbName, tableName);
-  return docs.length;
+  const tableKeys = await getTableKeys(dbName, tableName);
+  return tableKeys.length;
 }
 
 export async function findById(dbName: string, tableName: string, id: string) {
-  const docs = await find(dbName, tableName);
-  return docs.find((doc: any) => doc._id === id);
+  const doc = await AsyncStorage.getItem(getKey(dbName, tableName, id));
+  return doc ? JSON.parse(doc) : undefined;
 }
 
 export async function save(dbName: string, tableName: string, docs: any = []) {
-  const table = await getTable(dbName, tableName);
-  docs.forEach((doc: { _id: string }) => {
-    table.set(doc._id, doc);
-  });
-  await setTable(dbName, tableName, table);
-  return docs;
+  const kvPairs = docs.map(doc => [getKey(dbName, tableName, doc._id), JSON.stringify(doc)]);
+  await AsyncStorage.multiSet(kvPairs);
+  return kvPairs;
 }
 
 export async function removeById(dbName: string, tableName: string, id: string) {
-  const table = await getTable(dbName, tableName);
-  if (table.delete(id)) {
-    setTable(dbName, tableName, table);
+  const key = getKey(dbName, tableName, id);
+  const doc = await AsyncStorage.getItem(key);
```

Comment:
Do we need to check if it exists before removing it?

Reply:
Can we determine if the remove happened from the AsyncStorage.removeItem result?

Reply:
It doesn't seem so `removeItem` returns a `Promise<void>`.